Jasmine/Tetris/ViewModels/TetrisGameViewModel.swift

```diff
@@ -53,46 +67,38 @@ class TetrisGameViewModel {
         }
     }
 
-    /// Checks for and returns coordinates of matching phrase, searching by row-wise and return if found
-    /// otherwise continue with searching by column-wise
+    /// Checks for and returns coordinates of matching phrase, searching by row-wise then column-wise
     fileprivate func checkForMatchingPhrase() -> Set<Coordinate>? {
-        return checkForMatchingPhrase(byRow: true) ?? checkForMatchingPhrase(byRow: false)
-    }
-
-    /// Checks for and returns coordinates of matching phrase, search row/col-wise as specify by searchbyRow
-    /// Concatenate the words row by row or col by col to check if a phrase is contained in them
-    private func checkForMatchingPhrase(byRow searchByRow: Bool) -> Set<Coordinate>? {
-        var matchedCoordinates: Set<Coordinate> = []
-        let maxIndex = searchByRow ? Constants.Game.Tetris.rows : Constants.Game.Tetris.columns
-        for index in 0..<maxIndex {
-            var line = ""
-            if searchByRow {
-                for col in 0..<Constants.Game.Tetris.columns {
-                    line += getTileText(at: Coordinate(row: index, col: col))
-                }
-            } else {
-                for row in 0..<Constants.Game.Tetris.rows {
-                    line += getTileText(at: Coordinate(row: row, col: index))
+        let phraseLen = gameData.phrases.phraseLength
+        for row in 0..<Constants.Game.Tetris.rows - phraseLen {
+            for col in 0..<Constants.Game.Tetris.columns {
+                let phraseRange = col..<col + phraseLen
```

Comment:
I suggest you use `col..<(col + phraseLen)` here to prevent confusion. Also, this code is duplicated with the bottom one, can we do something about it?